src/test/java/org/apache/bcel/data/SWAP.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.bcel.data;
+
+import java.lang.reflect.Constructor;
+
+/*
+ * A test for SWAP instruction.
+ * To be compiled with Eclipse Compiler (ecj) with source/target 1.4.
```

Comment:
If that is really a requirement for this test fixture, then a precompiled class file should be in the test resources. Or am I missing something?

Reply:
It's indeed a possibility, but I'm not a fan of committing a class file in a repo, if we update the source, the class also needs to be updated and committed too, and also when there's a class file in a repo, one always wonders where it comes from, how it was generated, if it actually matches the source code, or if it can be trusted. If pulling a dependency is more of a problem, I can commit the class instead.

Reply:
Either the comment is wrong, or, the class should be in the repo. We already have plenty of class files in the repo, see src/test/resources. This component is unlike most since it works with these types of files explicitly, otherwise you would not want class files in a repo.